assets/js/data/cart/actions.js

```diff
@@ -258,6 +273,7 @@ export function* changeCartItemQuantity( cartItemKey, quantity ) {
  */
 export function* selectShippingRate( rateId, packageId = 0 ) {
 	try {
+		yield shippingRatesBeingSelected( true );
```

Comment:
so a question, can’t we rely on functions like `hasFinishedResolution` never understood when can we use them.

Reply:
The resolution functions from `wp.data` are only useful for `resolvers` attached to selectors. There is no automatic enhancement/detection of action generators that use the api. That might be a good thing to add to `wp.data` some day but for now if we need to track resolution state of action generators we need to implement our own state for that.